src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsEntityTypeDefinition.cs

```diff
@@ -112,7 +112,39 @@ private IEnumerable<IEdmStructuralProperty> ComputeDeclaredKey()
                 List<IEdmStructuralProperty> key = new List<IEdmStructuralProperty>();
                 foreach (CsdlPropertyReference keyProperty in this.entity.Key.Properties)
                 {
-                    IEdmStructuralProperty structuralProperty = this.FindProperty(keyProperty.PropertyName) as IEdmStructuralProperty;
+                    IEdmStructuralProperty structuralProperty = null;
+                    if (!string.IsNullOrEmpty(keyProperty.Alias))
+                    {
+                        string[] keyProperties = keyProperty.PropertyName.Split('/');
+                        for (int i = 0; i < keyProperties.Length; i++)
+                        {
+                            if (i == 0)
+                            {
+                                structuralProperty = this.FindProperty(keyProperties[i]) as IEdmStructuralProperty;
+                                i++;
+                            }
+                            else
+                            {
+                                structuralProperty = this.FindProperty(keyProperties[i]) as IEdmStructuralProperty;
+                                i++;

```

Comment:
It seems that we are actually counting by 2. It might be good to just have the `for` loop use `i += 2` instead so that the loop body is more clear